scripts/tablegen.py

```diff
@@ -58,73 +71,119 @@ def endOfTheList(context, lineNum, lastLineNum, totalNum):
-        else:
-            storekey = '.'.join(sk[:0]).lstrip('.')
-        
-        if  len(context[lastLineNum].lstrip()) != 0 and '#' != context[lastLineNum].lstrip()[0]:
-            isEnd, ValueList  = endOfTheList(context, lineNum, lastLineNum, totalNum)
-            if (isEnd == True):
-                keysplit = key.split('.')
-                for kv in keysplit:
-                    if kv != '':
-                        newkey = newkey + '.' + kv
-                newkey = newkey.lstrip('.')
-
-                ValueStr = (' ').join(ValueList)
-                if ValueStr:
-                    desc = ''
-                    prdict[storekey].append("| `%s` | `%s` | %s |" % (newkey, ValueStr, desc))
-                desc = ''
+    return ret_val
 
-                key = newkey
-                newkey = ''
+def process_helm_yaml(values_yaml, option):
+        ret_val = ''
+        storekey = ''
+        desc = ''
+        newkey = ''
+        whitespaces = 0
+        flag = 0
+        lineNum = 0
+        lastLineNum = 0
+        valueList = []
+        newConfigList = []
+
+        context = linecache.getlines(values_yaml)
+        totalNum = len(context)
+        lastLineNum = 0
+        key = option
+
+        for lineNum in range(0, totalNum):
+            if  context[lineNum].strip().startswith('- '):
+                pass
+            elif '#' in context[lineNum] and '#' == context[lineNum].lstrip()[0]:
+                if "Description: " in context[lineNum]:
+                    desc = context[lineNum].strip()
+            elif ':' in context[lineNum] and '#' != context[lineNum].lstrip()[0]:
+                lastLineNum = lineNum
+                if flag == 1:
+                    whitespaces = (len(context[lineNum]) - len(context[lineNum].lstrip())) / 2
+                    periods = key.count('.')
+                    if (option == ''):
+                        while (whitespaces <= periods):
+                            key = key.rstrip(string.ascii_letters[::-1] + string.digits + '_' + '-' + '/').rstrip('.')
+                            whitespaces += 1
+                    else:
+                        while (whitespaces < periods) :
+                            key = key.rstrip(string.ascii_letters[::-1] + string.digits + '_' + '-' + '/').rstrip('.')
+                            whitespaces += 1
+                        flag = 0
+
+                key = key + '.' + context[lineNum].split(':', 1)[0].strip()
+                isEnd, ValueList  = endOfTheList(context, lineNum, lastLineNum, totalNum)
+                if isEnd == True:
+                    flag = 1;
+
+            storekey = key
+            sk = storekey.split('.', 2)
+            if len(sk) > 1:
+                storekey = '.'.join(sk[:1]).lstrip('.')
+            else:
+                storekey = '.'.join(sk[:0]).lstrip('.')
+
+            #
+            # If we are processing the configurations options within the values.yaml under istio,
```

Comment:
how does this condition occur?

Reply:
I have included comments in the latest PR. This condition is used to check if the configuration option is in values.yaml under istio directory and if this option is a new option or processed before (for example, if this was found in the sub charts directory). The assumption here is that the configuration options under the sub charts directory has the latest values and we want to keep them and only process the configuration options that are only in the istio/values.yaml dir like 'global' option for example. 

Hope this helps. Thanks for the review

Reply:
The code currently (or did) work by replacing all contents between  `<!-- AUTO-GENERATED-START --` and `<!-- AUTO-GENERATED-END -->`.  We can't select out keys that are in the existing MD as the comments or default may have changed.  Can you take a look at ensuring this case works?  The code takes fractions of a second to run, so we aren't all that concerned about effeciency.

Reply:
> @sdake @linsun thank you for your kind comments and review. Appreciate it. I will make those changes you recommended in this PR and I can take a look at filling the description fields also. Is there a WI associated to that already or should we create one?

Can handle the description field as a separate PR. The existing issue for this is: https://github.com/istio/istio.io/issues/2566

Reply:
@sdake This code works exactly like you mentioned, i.e, replace everything in the markdown file between <!-- AUTO-GENERATED-START -- and <!-- AUTO-GENERATED-END -->.  That new if-else block processes the configuration options in the values.yaml under the istio/install/kubernetes/helm/istio directory. There are configuration options in there like:
- kiali : we want to make sure that the parameters for this option that will be populated in the index.md are from istio/install/kubernetes/helm/subcharts/kiali/values.yaml and not istio/install/kubernetes/helm/istio/values.yaml

- global: This does not have a corresponding directory under istio/install/kubernetes/helm/subcharts/ so we will get the parameters for this option from stio/install/kubernetes/helm/istio/values.yaml.

That was the thought process.